### PR TITLE
INT-4616 test pipeline syntax page

### PR DESCRIPTION
**Pull request by kellyrob99**
Enhance functional testing of the pipeline syntax page, which relies on ajax requests to a configured server to function correctly.

- [X ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [X ] Ensure that the pull request title represents the desired changelog entry
- [X ] Please describe what you did
- [X ] Link to relevant issues in GitHub or Jira
- [X ] Link to relevant pull requests, esp. upstream and downstream changes
- [ X] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/test/java/org/sonatype/nexus/ci/iq/IqServerMockUtility.groovy [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        "timestamp": "201807111516",
+        "build": "build-number"}""")))
+    givenThat(post(urlMatching('/api/v2/sourceControl.*'))
+        .willReturn(okJson("""{"stageTypeId": "stageTypeId",
+            "ownerId" : "",
+            "repositoryUrl": "",
+            "token": "",
+            "provider": ""
+            }""")))
+    givenThat(get(urlPathEqualTo('/rest/policy/stages'))
```
Comment (kellyrob99) [inline src/test/java/org/sonatype/nexus/ci/iq/IqServerMockUtility.groovy:88]: only new part here

File: src/test/java/org/sonatype/nexus/ci/iq/IqServerMockUtility.groovy [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        .willReturn(okJson('[{"stageTypeId":"develop","stageName":"Develop"},{"stageTypeId":"build","stageName":"Build"},{"stageTypeId":"stage-release","stageName":"Stage Release"},{"stageTypeId":"release","stageName":"Release"},{"stageTypeId":"operate","stageName":"Operate"}]'))
```
Comment (adrianpowell) [inline src/test/java/org/sonatype/nexus/ci/iq/IqServerMockUtility.groovy:89]: can this line be wrapped?

Comment (kellyrob99) [reply src/test/java/org/sonatype/nexus/ci/iq/IqServerMockUtility.groovy]: > can this line be wrapped?

6de49d3e5de8271d57c1c0eb730bd6bb99772fcf

**Review by adrianpowell — APPROVED** [review]
the new test looks like it's doing what's expected +1

**Review by collinpeters — APPROVED** [review]
+1

**Pull request merged**